File: interviews/stripe/request-replay-master/scala/src/main/scala/MyLog.scala
package com.stripe.interview

import spray.json.DefaultJsonProtocol

object MyLog extends DefaultJsonProtocol{
  case class Request(val url:String, val headers: Map[String,String], body:String, method:String)
  case class Response(val body:String, val headers: Map[String,String], code:Int)
  case class Pair(request:Request,response:Response)
  //only needed for the second question
  case class ExtractableResponse(`object`:String,id:String)

  implicit val responseFormat = jsonFormat3(Response.apply)
  implicit val requestFormat = jsonFormat4(Request.apply)
  implicit val pairFormat = jsonFormat2(Pair.apply)
  implicit val extractableResponse = jsonFormat2(ExtractableResponse.apply)
}

File: interviews/stripe/request-replay-master/java/src/main/java/com/stripe/interview/MyLog.java
package com.stripe.interview;

import java.util.Map;

public class MyLog {
	public MyRequest request;
	public MyResponse response;

	public MyRequest getRequest() {
		return request;
	}

	public MyResponse getResponse() {
		return response;
	}

	public static class MyResponse {
		public String body;
		public int code;

		public String getBody() {
			return body;
		}

		public int getCode() {
			return code;
		}
	}

	public static class MyRequest {
		public String url;
		public Map<String, String> headers;
		public String body;

		public String getUrl() {
			return url;
		}

		public Map<String, String> getHeaders() {
			return headers;
		}

		public String getBody() {
			return body;
		}
	}
}
